internal/controllers/gateway/route_utils_test.go

```diff
@@ -1200,4 +1390,234 @@ func Test_getSupportedGatewayForRoute(t *testing.T) {
 			})
 		}
 	})
+
+	t.Run("TLSRoute", func(t *testing.T) {
+		type expected struct {
+			gateway      types.NamespacedName
+			condition    metav1.Condition
+			listenerName string
+		}
+		tests := []struct {
+			name     string
+			route    *TLSRoute
+			expected []expected
+			objects  []client.Object
+			wantErr  bool
+		}{
+			{
+				name: "basic TLSRoute does get accepted because it is in supported kinds and there is a listener with TLS in passthrough mode",
```

Comment:
We need to change the unit tests if we follow the compatibility matrix with less restrictions: https://gateway-api.sigs.k8s.io/concepts/api-overview/#route-summary-table